corehq/apps/accounting/tasks.py

```diff
@@ -227,6 +235,19 @@ def send_subscription_reminder_emails(num_days, exclude_trials=True):
             subscription.send_ending_reminder_email()
 
 
+def send_subscription_reminder_emails_dimagi_contact(num_days):
+    today = datetime.date.today()
+    date_in_n_days = today + datetime.timedelta(days=num_days)
+    ending_subscriptions = (Subscription.objects
+                            .filter(is_active=True)
+                            .filter(date_end=date_in_n_days)
```

Comment:
> If celery goes down, the queue still gets populated

I don't think that's how it works. I think celery is the thing that manages the schedule and populates the queue. However, I may be wrong, and would be happy to be corrected. In any case, some part of our system is populating that queue, and that part of the system could be down when the condition `date_end=date_in_n_days` is matched, which means that the alert would not be sent. This is more of a design question: do we need to think about somehow tracking pending alerts over a wider date range (more than a single day) to prevent them from being skipped if the system has a hiccup at the wrong moment. Granted, the design does get a bit more tricky because you will likely need to keep track of whether a given alert has been sent or not.